### PR TITLE
DRIVERS-2805 Convert Documentation to Markdown

### PR DESCRIPTION
**Pull request by blink1073**
This will prepare us to host the docs on Pine.

### CODE REVIEW

File: docs/technical-design.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+## Definitions
+
+### META
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”,
+“SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this
+document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Terms
```
Comment (Jibola) [inline docs/technical-design.md:58]: Looks like this section previously had [indentations](https://github.com/mongodb-labs/drivers-atlas-testing/blob/master/docs/source/technical-design.rst#terms) for the terms?

I think it's fine if we just bolden the word and leave the other part as is.

File: docs/technical-design.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+Atlas Cluster
```
Comment (Jibola) [inline docs/technical-design.md:60]:
```suggestion
**Atlas Cluster**  
```

File: docs/technical-design.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Terms
+
+Atlas Cluster  
+Subordinate of an *Atlas Project* in Atlas’ Organizations and Projects
+hierarchy[^1], an Atlas Cluster is a set of nodes comprising a MongoDB
+deployment.
+
+Atlas Project
```
Comment (Jibola) [inline docs/technical-design.md:65]:
```suggestion
**Atlas Project**  
```

File: docs/technical-design.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+Subordinate of an *Atlas Project* in Atlas’ Organizations and Projects
+hierarchy[^1], an Atlas Cluster is a set of nodes comprising a MongoDB
+deployment.
+
+Atlas Project  
+Subordinate of an *Atlas Organization* in Atlas’ Organizations and
+Projects hierarchy[^2]. Each Atlas Organization can contain multiple
+Atlas Projects.
+
+Atlas Organization
```
Comment (Jibola) [inline docs/technical-design.md:70]:
```suggestion
**Atlas Organization**  
```

File: docs/technical-design.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+Atlas Project  
+Subordinate of an *Atlas Organization* in Atlas’ Organizations and
+Projects hierarchy[^2]. Each Atlas Organization can contain multiple
+Atlas Projects.
+
+Atlas Organization  
+The top-level entity in Atlas’ Organizations and Projects hierarchy[^3].
+
+Atlas Planned Maintenance Test
```
Comment (Jibola) [inline docs/technical-design.md:73]:
```suggestion
**Atlas Planned Maintenance Test**  
```

File: docs/technical-design.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+Atlas Organization  
+The top-level entity in Atlas’ Organizations and Projects hierarchy[^3].
+
+Atlas Planned Maintenance Test  
+An integration test that involves running a *Driver Workload* using a
+MongoDB Driver against an *Atlas Cluster* that is in the midst of
+applying a *Planned Maintenance Scenario*.
+
+Cluster Configuration Options
```
Comment (Jibola) [inline docs/technical-design.md:78]:
```suggestion
**Cluster Configuration Options**  
```

File: docs/technical-design.md [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+MongoDB Driver against an *Atlas Cluster* that is in the midst of
+applying a *Planned Maintenance Scenario*.
+
+Cluster Configuration Options  
+Umbrella term used to refer to all parameters that can be used to
+configure an *Atlas Cluster* via the *Atlas API*. There are two kinds of
+cluster configuration options - *Basic Configuration Options* and
+*Advanced Configuration Options*.
+
+Basic Configuration Options
```
Comment (Jibola) [inline docs/technical-design.md:84]:
```suggestion
**Basic Configuration Options**  
```

File: docs/technical-design.md [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+configure an *Atlas Cluster* via the *Atlas API*. There are two kinds of
+cluster configuration options - *Basic Configuration Options* and
+*Advanced Configuration Options*.
+
+Basic Configuration Options  
+Set of cluster configuration parameters that can be used to create a new
+*Atlas Cluster* via the Create One Cluster[^4] endpoint, or that can be
+applied to an existing cluster via the Modify One Cluster[^5] endpoint.
+
+Advanced Configuration Options
```
Comment (Jibola) [inline docs/technical-design.md:89]:
```suggestion
**Advanced Configuration Options**  
```

File: docs/technical-design.md [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+Set of cluster configuration parameters that can be used to create a new
+*Atlas Cluster* via the Create One Cluster[^4] endpoint, or that can be
+applied to an existing cluster via the Modify One Cluster[^5] endpoint.
+
+Advanced Configuration Options  
+Set of cluster configuration parameters that be applied to an existing
+*Atlas Cluster* via the Modify Advanced Configuration Options for One
+Cluster[^6] endpoint.
+
+Cluster State
```
Comment (Jibola) [inline docs/technical-design.md:94]:
```suggestion
**Cluster State**  
```

File: docs/technical-design.md [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+Advanced Configuration Options  
+Set of cluster configuration parameters that be applied to an existing
+*Atlas Cluster* via the Modify Advanced Configuration Options for One
+Cluster[^6] endpoint.
+
+Cluster State  
+The state of an *Atlas Cluster* as advertised by the “stateName” field
+in the response of the Get One Cluster[^7] endpoint.
+
+Test Scenario File
```
Comment (Jibola) [inline docs/technical-design.md:98]:
```suggestion
**Test Scenario File**  
```

File: docs/technical-design.md [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+Cluster State  
+The state of an *Atlas Cluster* as advertised by the “stateName” field
+in the response of the Get One Cluster[^7] endpoint.
+
+Test Scenario File  
+A file containing a language and driver-agnostic description of a single
+*Atlas Planned Maintenance Test*. Each file specifies a *Planned
+Maintenance Scenario* and the associated *Driver Workload*.
+
+Planned Maintenance Scenario
```
Comment (Jibola) [inline docs/technical-design.md:103]:
```suggestion
**Planned Maintenance Scenario**  
```

File: docs/technical-design.md [hunk 12]
```diff
@@ -0,0 +1,335 @@
...
+A file containing a language and driver-agnostic description of a single
+*Atlas Planned Maintenance Test*. Each file specifies a *Planned
+Maintenance Scenario* and the associated *Driver Workload*.
+
+Planned Maintenance Scenario  
+Description of a plan to modify the *Cluster Configuration Options* of a
+running *Atlas Cluster*. A maintenance plan is described fully by an
+associated set of initial and final *Cluster Configuration Options*.
+
+Driver Workload
```
Comment (Jibola) [inline docs/technical-design.md:108]:
```suggestion
**Driver Workload**  
```

File: docs/technical-design.md [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+
+Planned Maintenance Scenario  
+Description of a plan to modify the *Cluster Configuration Options* of a
+running *Atlas Cluster*. A maintenance plan is described fully by an
+associated set of initial and final *Cluster Configuration Options*.
+
+Driver Workload  
+A language-agnostic description of MongoDB driver operations.
+
+Workload Executor
```
Comment (Jibola) [inline docs/technical-design.md:111]:
```suggestion
**Workload Executor**  
```

File: docs/technical-design.md [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+
+Driver Workload  
+A language-agnostic description of MongoDB driver operations.
+
+Workload Executor  
+A user-implemented, driver-specific script responsible for parsing a
+*Driver Workload* and translating it into actual driver operations that
+are run against the test cluster.
+
+Test Orchestrator
```
Comment (Jibola) [inline docs/technical-design.md:116]:
```suggestion
**Test Orchestrator**  
```

File: docs/technical-design.md [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+Workload Executor  
+A user-implemented, driver-specific script responsible for parsing a
+*Driver Workload* and translating it into actual driver operations that
+are run against the test cluster.
+
+Test Orchestrator  
+User-facing command-line utility that accepts a *Workload Executor* and
+a *Test Scenario File* and runs an *Atlas Planned Maintenance Test*.
+
+Atlas API
```
Comment (Jibola) [inline docs/technical-design.md:120]:
```suggestion
**Atlas API**  
```

**Review by Jibola — CHANGES_REQUESTED** [review]
Looks great! I found some formatting changes but this was a much needed conversion. Thank you!

File: docs/technical-design.md (hunk 1 above)
Comment (blink1073) [reply docs/technical-design.md:58]: Yeah bold sounds good

File: docs/technical-design.md [hunk 16]
```diff
@@ -0,0 +1,335 @@
...
+Approved
+
+Type  
+Process
+
+Minimum Server Version  
+N/A
+
+Last Modified  
+January 28, 2020
```
[the remark this answers is not in the document]
Comment (blink1073) [reply docs/technical-design.md:32]: GFM doesn't support tables without a header, I think that's how this one ended up awkwardly formatted.  I'll add bolds here too.

**Review by Jibola — APPROVED** [review]

**Pull request merged**